packages/core/src/routes/interaction/verifications/mandatory-user-profile-validation.ts

```diff
@@ -94,71 +95,85 @@ const validateRegisterMandatoryUserProfile = (profile?: Profile) => {
+): Promise<{
+  missingProfileSet: Set<MissingProfile>;
+  interaction: MandatoryProfileValidationInteraction;
+  registeredSocialIdentity?: { email?: string; phone?: string };
+}> => {
   const { identifiers = [], profile } = interaction;
 
   const socialIdentifier = identifiers.find(
     (identifier): identifier is SocialIdentifier => identifier.key === 'social'
   );
 
   if (!socialIdentifier) {
-    return [missingProfile, interaction];
+    return { missingProfileSet, interaction };
   }
 
   const {
     userInfo: { email, phone },
   } = socialIdentifier;
 
+  // Email Required
   if (
-    (missingProfile.has(MissingProfile.email) || missingProfile.has(MissingProfile.emailOrPhone)) &&
-    email &&
-    // Email taken
-    !(await userQueries.hasUserWithEmail(email))
+    (missingProfileSet.has(MissingProfile.email) ||
+      missingProfileSet.has(MissingProfile.emailOrPhone)) &&
+    email
   ) {
-    missingProfile.delete(MissingProfile.email);
-    missingProfile.delete(MissingProfile.emailOrPhone);
+    // Email taken
+    if (await userQueries.hasUserWithEmail(email)) {
+      return { missingProfileSet, interaction, registeredSocialIdentity: { email } };
+    }
 
-    // Assign social verified email to the interaction
-    return [
-      missingProfile,
-      {
+    // Assign social verified email to the interaction and remove from missingProfile
+    missingProfileSet.delete(MissingProfile.email);
+    missingProfileSet.delete(MissingProfile.emailOrPhone);
+
+    return {
+      missingProfileSet,
+      interaction: {
         ...interaction,
         identifiers: mergeIdentifiers({ key: 'emailVerified', value: email }, identifiers),
         profile: {
           ...profile,
           email,
         },
       },
-    ];
+    };
   }
 
+  // Phone required
   if (
-    (missingProfile.has(MissingProfile.phone) || missingProfile.has(MissingProfile.emailOrPhone)) &&
-    phone &&
-    // Phone taken
-    !(await userQueries.hasUserWithPhone(phone))
+    (missingProfileSet.has(MissingProfile.phone) ||
+      missingProfileSet.has(MissingProfile.emailOrPhone)) &&
+    phone
   ) {
-    missingProfile.delete(MissingProfile.phone);
-    missingProfile.delete(MissingProfile.emailOrPhone);
+    // Phone taken
+    if (await userQueries.hasUserWithPhone(phone)) {
+      return { missingProfileSet, interaction, registeredSocialIdentity: { phone } };
+    }
 
-    // Assign social verified phone to the interaction
-    return [
-      missingProfile,
-      {
+    // Assign social verified phone to the interaction and remove from missingProfile
+    missingProfileSet.delete(MissingProfile.phone);
+    missingProfileSet.delete(MissingProfile.emailOrPhone);
+
+    return {
+      missingProfileSet,
+      interaction: {
         ...interaction,
         identifiers: mergeIdentifiers({ key: 'phoneVerified', value: phone }, identifiers),
         profile: {
           ...profile,
           phone,
         },
       },
-    ];
+    };
```

Comment:
I think I asked this question in a previous PR but I'm not sure about the answer. What if the social account has both the email and phone? It seems our code can take either email or phone but not btoh of them at a time. Is this by design?